File: frontend/src/objects/customer/apiTypes/AddCustomerAddressRequest.ts
// Business note: service protocol DTO shared with backend apiTypes; keep field names and value object types aligned.
import type {
  AddressLabel,
  AddressText,
  DeliveryCoordinate,
} from '@/objects/core/SharedObjects'

export type AddCustomerAddressRequest = {
  label: AddressLabel
  address: AddressText
  location: DeliveryCoordinate
}

File: backend/src/main/scala/services/customer/objects/apiTypes/AddCustomerAddressRequest.scala
package services.customer.objects.apiTypes

// Business note: protocol DTO shared with the frontend; keep field names and value object types mirrored across stacks.
import services.customer.objects.*

import system.objects.given

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.*
import system.objects.*

final case class AddCustomerAddressRequest(
    label: AddressLabel,
    address: AddressText,
    location: CustomerLocation,
)
object AddCustomerAddressRequest:
  given Encoder[AddCustomerAddressRequest] = deriveEncoder
  given Decoder[AddCustomerAddressRequest] = deriveDecoder
